src/library_syscall.js

```diff
@@ -973,7 +974,12 @@ var SyscallsLibrary = {
     off <<= 12; // undo pgoffset
     var ptr;
     var allocated = false;
-    if (fd === -1) {
+
+    if ((flags & {{{ cDefine('MAP_FIXED') }}}) !== 0 && (addr % PAGE_SIZE) !== 0) {
```

Comment:
I'm not sure what this is checking? A comment could be helpful - perhaps in the other places too.

Reply:
Added comment in https://github.com/emscripten-core/emscripten/pull/8633/commits/43fcce7707af84dfe995df60a4fd80e8456e6fad